arrow/util.py

```diff
@@ -14,7 +17,9 @@ def is_timestamp(value):
     if isinstance(value, bool):
         return False
     if not (
-        isinstance(value, int) or isinstance(value, float) or isinstance(value, str)
+        isinstance(value, numbers.Integral)
```

Comment:
Covers python 2.7?

Reply:
Yeah! Turns out the function as-is fails on larger millisecond or microsecond timestamps since Python 2.7 separated the long and int types (long was used for 64 bit integers). All integers are now longs in Python 3+, so once we drop <3.5, we can change this back to `isinstance(value, int)`.